lectures/cross_section.md

```diff
@@ -0,0 +1,859 @@
+```{figure} /_static/lecture_specific/cross_section/light_heavy_fig1.png
+
+```
+
+In the top subfigure, the standard deviation of the normal distribution is 2,
+and the draws are clustered around the mean.
+
+In the middle subfigure, the standard deviation is increased to 12 and, as
+expected, the amount of dispersion rises.
+
+The bottom subfigure, with the Cauchy draws, shows a different pattern: tight
+clustering around the mean for the great majority of observations, combined
+with a few sudden large deviations from the mean.
+
+This is typical of a heavy-tailed distribution.
+
+
+## Heavy Tails in Economic Cross-Sectional Distributions
+
+TODO 
+
+- Shu please add data and plots.
+- Please use empirical CCDF, as in the macro dynamics book
+
+
+
+TODO Review exercises below --- are they all too hard for undergrads or should
+we keep some of them.
+
+```{code-cell} ipython3
+def extract_wb(varlist=['NY.GDP.MKTP.CD'], c='all', s=1900, e=2021):
+    df = wb.download(indicator=varlist, country=c, start=s, end=e).stack().unstack(0).reset_index()
+    df = df.drop(['level_1'], axis=1).set_index(['year']).transpose()
+    return df
+```
+
+```{code-cell} ipython3
+def empirical_ccdf(data, 
+                   ax, 
+                   aw=None,   # weights
+                   label=None,
+                   xlabel=None,
+                   add_reg_line=False, 
+                   title=None):
+    """
+    Take data vector and return prob values for plotting.
+    Upgraded empirical_ccdf
+    """
+    y_vals = np.empty_like(data, dtype='float64')
+    p_vals = np.empty_like(data, dtype='float64')
+    n = len(data)
+    if aw is None:
+        for i, d in enumerate(data):
+            # record fraction of sample above d
+            y_vals[i] = np.sum(data >= d) / n
+            p_vals[i] = np.sum(data == d) / n
+    else:
+        fw = np.empty_like(aw, dtype='float64')
+        for i, a in enumerate(aw):
+            fw[i] = a / np.sum(aw)
+        pdf = lambda x: interp(data, fw, x)
+        data = np.sort(data)
+        j = 0
+        for i, d in enumerate(data):
+            j += pdf(d)
+            y_vals[i] = 1- j
+
+    x, y = np.log(data), np.log(y_vals)
+    
+    results = sm.OLS(y, sm.add_constant(x)).fit()
+    b, a = results.params
+    
+    kwargs = [('alpha', 0.3)]
+    if label:
+        kwargs.append(('label', label))
+    kwargs = dict(kwargs)
+
+    ax.scatter(x, y, **kwargs)
+    if add_reg_line:
+        ax.plot(x, x * a + b, 'k-', alpha=0.6, label=f"slope = ${a: 1.2f}$")
+    if not xlabel:
+        xlabel='log value'
+    ax.set_xlabel(xlabel, fontsize=12)
+    ax.set_ylabel("log prob.", fontsize=12)
+        
+    if label:
+        ax.legend(loc='lower left', fontsize=12)
+        
+    if title:
+        ax.set_title(title)
+        
+    return np.log(data), y_vals, p_vals
+```
+
+### GDP
+
+```{code-cell} ipython3
+df_gdp1 = extract_wb(varlist=['NY.GDP.MKTP.CD']) # gdp for all countries from 1960 to 2022
```

Comment:
Can you please make this for just one year -- the latest available?

I don't think it makes sense to pool this data across years.